web_ir_actions_act_window_message/static/src/js/web_ir_actions_act_window_message.js

```diff
@@ -1,50 +1,89 @@
 
-openerp.web_ir_actions_act_window_message = function(instance)
-{
-    instance.web.ActionManager.include({
-        ir_actions_act_window_message: function(action, options)
-        {
-            var dialog = new instance.web.Dialog(
+odoo.define('web.web_ir_actions_act_window_message', function (require) {
+    "use strict";
+
+    var ActionManager = require('web.ActionManager'),
+        core = require('web.core'),
+        _ = require('_'),
+        Model = require('web.Model'),
+        Dialog = require('web.Dialog');
+
+    var _t = core._t;
+
+    ActionManager.include({
+        ir_actions_act_window_message: function(action, options){
+
+            var self = this,
+                buttons = [];
+
+            if(action.close_button_title !== false)
+            {
+                buttons.push({
+                    text: action.close_button_title || _t('Close'),
+                    click: function() {
+                        // refresh the view before closing the dialog
+                        self.inner_widget.active_view
+                            .controller.recursive_reload();
+                        dialog.close()
+                    },
+                    oe_link_class: 'oe_highlight',
+                })
+            }
+
+            var dialog = new Dialog(
                 this,
-                {
+                _.extend({
                     size: 'medium',
                     title: action.title,
-                    buttons: [
-                        {
-                            text: instance.web._t('Close'),
-                            click: function() { dialog.close() },
-                            oe_link_class: 'oe_highlight',
-                        },
-                    ],
-                },
-                jQuery(instance.web.qweb.render(
-                    'web_ir_actions_act_window_message',
-                    {
-                        'this': this,
-                        'action': action,
-                    }))
-            )
+                    $content: $('<div>', {
+                        text: action.message,
+                    }),
+                    buttons: buttons.concat(
+                        this.ir_actions_act_window_message_get_buttons(
+                            action, function() { dialog.close() })
+                    ),
+                }, options)
+            );
             return dialog.open();
         },
+        ir_actions_act_window_message_get_buttons: function(action, close_func)
+        {
+            // return an array of button definitions from action
+            var self = this;
+            return _.map(action.buttons || [], function(button_definition)
+            {
+                return {
+                    text: button_definition.name || 'No name set',
+                    oe_link_class: button_definition.oe_link_class ||
+                                   'oe_highlight',
```

Comment:
@zakiuu I just figured out that Odoo uses "classes" attribute instead of "oe_link_class" for dialog's buttons. "oe_highlight" class doesn't exists since 9.0.

```javascript
    set_buttons: function(buttons) {
        var self = this;

        self.$footer.empty();

        _.each(buttons, function(b) {
            var text = b.text || "";
            var classes = b.classes || ((buttons.length === 1)? "btn-primary" : "btn-default");
            var $b = $(QWeb.render('WidgetButton', { widget : { string: text, node: { attrs: {'class': classes} }}}));
```